### PR TITLE
replace sha1 with sha512

### PR DESCRIPTION
**Pull request by zshihang**
**What type of PR is this?**

/kind bug
/kind failing-test

**What this PR does / why we need it**:
sha1 is no longer available in releases. 

**Special notes for your reviewer**:
this PR unblocks the reviving of upgrade tests. see https://github.com/kubernetes/kubernetes/issues/94487

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/gci/configure.sh [hunk 1]
```diff
@@ -197,8 +197,8 @@ function remount-flexvolume-directory {
 
 function install-gci-mounter-tools {
   CONTAINERIZED_MOUNTER_HOME="${KUBE_HOME}/containerized_mounter"
-  local -r mounter_tar_sha="${DEFAULT_MOUNTER_TAR_SHA}"
```
Comment (liggitt) [inline cluster/gce/gci/configure.sh:200]: avoid renaming if possible

**Comment by liggitt** [comment]
one nit, then lgtm

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -1144,7 +1144,7 @@ NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS: $(yaml-quote "${NODE_PROBLEM_DETECTOR_CUSTOM
 CNI_STORAGE_URL_BASE: $(yaml-quote "${CNI_STORAGE_URL_BASE:-}")
 CNI_TAR_PREFIX: $(yaml-quote "${CNI_TAR_PREFIX:-}")
 CNI_VERSION: $(yaml-quote "${CNI_VERSION:-}")
-CNI_SHA1: $(yaml-quote "${CNI_SHA1:-}")
```
Comment (liggitt) [inline cluster/gce/util.sh:1147]: I'm not positive whether populating CNI_SHA1 is still required or not to support skewed nodes (e.g. a 1.20 API server and 1.18 node) ... @karan, any idea?

File: cluster/gce/config-default.sh [hunk 3]
```diff
@@ -307,7 +307,7 @@ NODE_PROBLEM_DETECTOR_TAR_HASH="${NODE_PROBLEM_DETECTOR_TAR_HASH:-}"
 NODE_PROBLEM_DETECTOR_RELEASE_PATH="${NODE_PROBLEM_DETECTOR_RELEASE_PATH:-}"
 NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS="${NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS:-}"
 
-CNI_SHA1="${CNI_SHA1:-}"
```
Comment (liggitt) [inline cluster/gce/config-default.sh:310]: not sure if we need to keep setting the SHA1 hash vars and set SHA512 in parallel until the oldest node we support has inherited the SHA512 checking code.

**Comment by liggitt** [comment]
assign @zmerlynn for control plane config
assign @karan for node config and skew questions

File: cluster/gce/gci/configure.sh (hunk 1 above)
Comment (zshihang) [reply cluster/gce/gci/configure.sh:200]: done

File: cluster/gce/util.sh (hunk 2 above)
Comment (karan) [reply cluster/gce/util.sh:1147]: afaik that should not matter, this script should not depend on the apiserver version

**Review by karan — APPROVED** [review]

**Comment by liggitt** [comment]
cc @yliaog for the windows script change

**Comment by liggitt** [comment]
looks reasonable to me, would like lgtm from karan/yliaog

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 4]
```diff
@@ -350,14 +350,14 @@ function DownloadAndInstall-AuthPlugin {
   }
 
   if (-not ($kube_env.ContainsKey('EXEC_AUTH_PLUGIN_LICENSE_URL') -and
-            $kube_env.ContainsKey('EXEC_AUTH_PLUGIN_SHA1') -and
+            $kube_env.ContainsKey('EXEC_AUTH_PLUGIN_HASH') -and
```
Comment (yliaog) [inline cluster/gce/windows/k8s-node-setup.psm1:353]: is there going to be a separate cl to populate kube-env with EXEC_AUTH_PLUGIN_HASH? currently EXEC_AUTH_PLUGIN_SHA1 is in kube-env, but EXEC_AUTH_PLUGIN_HASH is not.

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 5]
```diff
@@ -958,13 +958,13 @@ function Install_Cni_Binaries {
   $release_url = "${env:WINDOWS_CNI_STORAGE_PATH}/${env:WINDOWS_CNI_VERSION}/"
   $tgz_url = ($release_url +
               "cni-plugins-windows-amd64-${env:WINDOWS_CNI_VERSION}.tgz")
-  $sha_url = ($tgz_url + ".sha1")
-  MustDownload-File -URLs $sha_url -OutFile $tmp_dir\cni-plugins.sha1
-  $sha1_val = ($(Get-Content $tmp_dir\cni-plugins.sha1) -split ' ',2)[0]
+  $sha_url = ($tgz_url + ".sha512")
+  MustDownload-File -URLs $sha_url -OutFile $tmp_dir\cni-plugins.sha512
```
Comment (yliaog) [inline cluster/gce/windows/k8s-node-setup.psm1:962]: gsutil ls -l gs://gke-release/cni-plugins/v0.8.5-gke*

currently .sha512 is not available, only .sha1

Comment (zshihang) [reply cluster/gce/windows/k8s-node-setup.psm1:962]: is there a script to generate the sha1 files? or manually did it? i'll generate sha512 files.

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 4 above)
Comment (zshihang) [reply cluster/gce/windows/k8s-node-setup.psm1:353]: yes

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 5 above)
Comment (yliaog) [reply cluster/gce/windows/k8s-node-setup.psm1:962]: sha1 files are there, sha512 is missing currently

Comment (zshihang) [reply cluster/gce/windows/k8s-node-setup.psm1:962]: i meant how those sha1 files are created. via a script or a command? i don't have permission to `gsutil cp` to the bucket.

Comment (pjh) [reply cluster/gce/windows/k8s-node-setup.psm1:962]: From https://github.com/kubernetes/kubernetes/blob/e99dbbde62015c15b516cdf39edb8d32fcd71729/cluster/gce/config-common.sh#L143, the bucket is `k8s-artifacts-cni`.

`gsutil ls -l gs://k8s-artifacts-cni/release/v0.8.7/cni-plugins-windows*`

shows that the sha512 files are present. I'm not sure why the `gke-release` bucket is different, but for this PR this change looks good.

Comment (yliaog) [reply cluster/gce/windows/k8s-node-setup.psm1:962]: replied in bug

**Review by pjh — APPROVED** [review]

**Comment by pjh** [comment]
/test pull-kubernetes-e2e-windows-gce

This presubmit will check that a cluster with Windows nodes will come up. The presubmit is flaky so it shouldn't be non-blocking, but as long as the cluster is created successfully and the tests run, it will show that this PR hasn't broken Windows nodes.

Thanks for the changes!

**Comment by pjh** [comment]
This looks fine - the tests ran and the node [startup log](https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/94975/pull-kubernetes-e2e-windows-gce/1313181116894547968/artifacts/e2e-0889c811e8-95646-windows-node-group-0t59/) shows that the Windows nodes came up successfully still. Don't worry about the failed test cases.

/lgtm

> @zshihang: The following test **failed**, say `/retest` to rerun all failed tests:
> 
> Test name	Commit	Details	Rerun command
> pull-kubernetes-e2e-windows-gce	[e99dbbd](https://github.com/kubernetes/kubernetes/commit/e99dbbde62015c15b516cdf39edb8d32fcd71729)	[link](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/94975/pull-kubernetes-e2e-windows-gce/1313181116894547968)	`/test pull-kubernetes-e2e-windows-gce`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=94975). [Your PR dashboard](https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3Azshihang). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open) when you hit one in your PR.

File: cluster/gce/gci/configure.sh [hunk 6]
```diff
@@ -126,9 +126,9 @@ function validate-hash {
   local -r file="$1"
   local -r expected="$2"
 
-  actual=$(sha1sum ${file} | awk '{ print $1 }') || true
+  actual=$(sha512sum ${file} | awk '{ print $1 }') || true
```
Comment (liggitt) [inline cluster/gce/gci/configure.sh:129]: should this check if the passed in hash matches the sha1 or sha512sum of the file?

Comment (zshihang) [reply cluster/gce/gci/configure.sh:129]: no, this script is bundled with the releases and the callers of this function will not pass sha1 in the future.
